### DISCUSSION TITLE
LowRes roles - substream resolution very low?

### DISCUSSION THREAD

**Discussion by asaupp** [blakeblackshear/frigate#1636] [Ask A Question] [answered]
Hi,

I'd be interessted in feedback. I'm running a number of cameras from different vendors. All have a VERY low resolution on the substream. I'd really love to reduce CPU utilization on my j4105, as its running very close to 100% even though I'm using a USB Coral. That is with 2x4K, 1x5MP and 1xFullHD cameras.

Is that me picking the wrong cameras - or is anybody running successfully with such lowres images? I wasn't impressed from detection, when trying.

Dahua IPC-HDW2831T-ZS-S2: 
  3840x 2160 vs. 704x576

Dahua IPC-HDW2831T-ZS-S2: 
   3840x 2160 vs. 704x576

ReoLink : 
  2560x1920 vs. 640x480

Foscam: 
   1920x1080 vs. 720p

**Comment by shred86** [comment]
Something doesn’t seem right as I’m running an Intel Celeron N3450 (6W TDP) which is even less powerful than the J4105, and my CPU usage sits around 25-30% running 4x 1280x720 substream for detect (in addition to 10 other docker containers). Some things to check:

1) Do you have ffpmeg hardware acceleration setup? For your CPU, this should be in the config.yml file:

````
ffmpeg:
  hwaccel_args:
    - -hwaccel
    - vaapi
    - -hwaccel_device
    - /dev/dri/renderD128
    - -hwaccel_output_format
    - yuv420p
 ````

2) What `fps` do you have set for detect? 5fps is typically good enough and is what I’m running.

3) Is there anything in your camera view that is constantly moving (e.g. trees, bushes, etc.)? If so, setting up a mask around those objects will help reduce CPU usage.

**Comment by asaupp** [comment]
@shred86 thanks for your informative response!

Motion maskes reduced frigate load by ~35% average -still frigate consumes more than 2 of the four cores. Yes, 5fps in camera + detect configured, HW acceleration is set. 
The current system load works for me / the other containers - but maybe there is more potential? Your proposal was low effort, high gain.

The Dahua really offers 4K (which is 9x of your sub stream resolution) and 704 × 576 substream (max, but only) which in my noob opinion is just not enough for proper detection.
https://www.dahuasecurity.com/products/productDetail/39027

Are there ways to figure how time's spend in frigate - or other potentials that I could investigate?

[Sample: CPU utilization, proc list, docker stats](https://ibb.co/rGJY8hr)

:-) 
Alex

**Comment by blakeblackshear** [comment]
It will be much easier to help if you provide your config file. The best way to see what is happening is to watch the debug feed for your cameras in the UI with motion boxes and regions turned on. It can also be helpful to toggle detection off for all your cameras to see how much CPU is used simply for decoding. That will tell you how much more is being used for detection.

**Comment by asaupp** [comment, marked as answer]
The debug feed doesn't show a lot of "detection fps" since the motion masks where applied.
CPU is still 1,5-2 Cores without detection / recording enabled (instead of 1,8-2,8) - so I assume the input video resolution is the main issue. 

Closing the dicussion - thanks!
